File: Loan_Portfolio/Models/DocDetail.cs
using System;
using System.Collections.Generic;
using Loan_Portfolio.Models;
namespace Loan_Portfolio.Models
{
	public class AOContact
	{

		public string FullName { get; set; }
		public string Role { get; set; }
		public string Contact { get; set; }
	}
	public class CommitmentModel
	{

		public string Facility { get; set; }
		public string Currency { get; set; }
		public int Amount { get; set; }


	}

	public class EventPermit
	{
		public varClauseRef<string> TransferOtherLend { get; set; }
		public varClauseRef<string> DisclosureOtherLend { get; set; }
		public varClauseRef<string> TransferBankNFinancialInst { get; set; }
		public varClauseRef<string> DisclosureBankNFinancialInst { get; set; }

		//e
		public varClauseRef<string> TransferAnyEntitMakesLoans { get; set; }

		//f
		public varClauseRef<string> DisclosureAnyEntitMakesLoans { get; set; }
		public varClauseRef<string> AnyRequirPotentTransfereeRegulate { get; set; }

	}


	public class FacilityModel
	{
		public string Type { get; set; }
		public string Maturity { get; set; }
		public string Facility { get; set; }
		public string Currency { get; set; }
		public int Amount { get; set; }

	}
	public class SecurityModel
	{

		public string p1 { get; set; }
		public string p2 { get; set; }
		public string p3 { get; set; }
		public string p4 { get; set; }
		public string p5 { get; set; }
		public string Comment { get; set; }

	}
	public class varClauseRef<T>
	{
		public string ClauseRef { get; set; }
		public T OtherVar { get; set; }
	}

	public class varString<T>
	{
		public string STRING { get; set; }
		public varClauseRef<T> Bool { get; set; }
	}
	public class varNum<T>
	{
		public float Number { get; set; }
		public varClauseRef<T> Bool { get; set; }
	}
	public class FinancialCovenantsModel
	{
		public varClauseRef<bool> debtLeverage { get; set; }
		public varClauseRef<bool> interestCover { get; set; }
		public varClauseRef<bool> CashflowCover { get; set; }
		public varClauseRef<bool> tangibleNetWorth { get; set; }
	}


	public class ObligationCauseAgentModel
	{
		public varClauseRef<bool> DocByBorrowToCreditors { get; set; }
		public varClauseRef<bool> DetailLitArbAdminProcess { get; set; }
		public varClauseRef<bool> NotificAnyDefault { get; set; }
		public varClauseRef<bool> FurtherInfoRequestFinConBusOp { get; set; }
		public varClauseRef<bool> NotificChngeOwnerOfBorrow { get; set; }
		public varClauseRef<bool> NotificChngeConstDocObligor { get; set; }
		//ones with day input
		public varNum<bool> ConsAnnualFinancialStatement { get; set; }
		public varNum<bool> ConsSemiAnnualFinancialStatement { get; set; }
		public varNum<bool> ConsQuartAnnualFinancialStatement { get; set; }
		public varNum<bool> ConsMonthAnnualFinancialStatement { get; set; }
		public varNum<bool> UnConsAnnualFinancialStatement { get; set; }
		public varNum<bool> UnConsSemiAnnualFinancialStatement { get; set; }
		public varNum<bool> UnConsQuartAnnualFinancialStatement { get; set; }
		public varNum<bool> UnConsMonthAnnualFinancialStatement { get; set; }
	}


	public class FormIdentificationModel
	{

		public string Principle_Borrow { get; set; }
		public string AllenOveryCorM_Num { get; set; }
		public string Relevant_Facility { get; set; }
	}

	public class SettingsModel
	{
		public string Docname { get; set; }
		public string status { get; set; }
	}
	public class DocDetail
	{
		
		//settings
		public SettingsModel settings { get; set; }

		//security
		public SecurityModel Security { get; set; }


		//1.1
		public FormIdentificationModel FormIdentification { get; set; }
		//1.2
		public List<bool> Common_Venders { get; set; }
		public string AdditionalLenders { get; set; }
		public string final_report_dist_list { get; set; }
		//1.3
		public string PSponsor { get; set; }
		//1.4
		public string OrigDate { get; set; }
		//1.5
		public string Date { get; set; }
		//1.6
		public List<CommitmentModel> Commit { get; set; }
		//1.7
		public List<AOContact> Contacts { get; set; }
		//1.8

		public bool restricted { get; set; }


		//2.1
		public FacilityModel Facilities { get; set; }
		//2.2 
		public varClauseRef<string> PurposeClause { get; set; }
		//2.3
		public string consentlvl { get; set; }
		//2.4
		public string extension { get; set; }
		//2.5
		public varClauseRef<bool> CreditToInterestRate { get; set; }
		//2.6
		public varClauseRef<bool> DisposeMandatorPrepayment { get; set; }
		//2.7
		public varClauseRef<bool> ChangeMandatorPrepayment { get; set; }
		//2.8
		public varClauseRef<bool> CreditToTax_Gross_up { get; set; }
		//2.9
		public varClauseRef<bool> CreditToNegPledge { get; set; }
		//2.10
		public varClauseRef<bool> CreditToRestriction_IncurrringIndebtedness { get; set; }
		//2.11
		public FinancialCovenantsModel FinancialCovenants { get; set; }
		//2.12
		public varClauseRef<bool> MoreFinancialCovenants { get; set; }
		//2.13
		public ObligationCauseAgentModel ObligationCause { get; set; }
		//2.14
		public bool IsFailPayEvent { get; set; }
		//2.15
		public varClauseRef<bool> MaterialAdverseChangeEvent { get; set; }
		//2.16
		public varClauseRef<string> ProvCrossDefault_CrossAccel { get; set; }
		//2.17
		public varString<string> CrossDefOrAccelThresh { get; set; }
		//2.18
		public varClauseRef<bool> CAgrContClauseFavBank { get; set; }
		//2.19
		public string GovLawPrincCAgr { get; set; }
		//2.20
		public varClauseRef<bool> CAPArbitration { get; set; }


		//3.1
		public varClauseRef<bool> VoteOnPerameter { get; set; }
		//3.2
		public varClauseRef<string> LenderSplitCommitLoansVote { get; set; }
		//3.3
		public varClauseRef<string> SplitVoteDetail { get; set; }
		//3.4
		public varClauseRef<bool> HedgeBeforeEnf { get; set; }
		//3.5
		public varClauseRef<bool> HedgeAfterEnf { get; set; }


		//4.1
		public varClauseRef<string> ObligorNameJurJob { get; set; }
		//4.2
		public varClauseRef<bool> GuarantorsGuarantAllBorrow { get; set; }
		//4.3
		public varClauseRef<bool> DoesAgreeAllowAddBorrow { get; set; }
		//4.4
		public varClauseRef<string> ConsLvlAdditionBorrowersNewJur { get; set; }
		//4.5
		public varClauseRef<bool> DoesAgreeAllowAddGuarantors { get; set; }
		//4.6
		public varClauseRef<string> ConsLvlAdditionBorrowersExstJur { get; set; }
		//4.7
		public varClauseRef<string> ConsLvlAdditionGuarantorsExstJur { get; set; }
		//4.8
		public varClauseRef<string> ConsLvlAdditionGuarantorsNewJur { get; set; }

		//5.1
		public bool IsSecurity { get; set; }
		//5.2
		public string SecurityDescrip { get; set; }
		//5.3
		public string LawsGovSecurity { get; set; }
		//5.4
		public bool IsIntercreditorAgreement { get; set; }
		//5.5
		public bool SecurityBenefitHedge { get; set; }
		//5.6
		public bool IsOriginalDocProvidedToSecurityTrust { get; set; }


		//6.1,6.2
		public List<EventPermit> DefaultnoDefault { get; set; }
		//6.3
		public varClauseRef<bool> PotentTransfereeRequirEntNondisclosureAgree { get; set; }
		//6.4
		public varClauseRef<bool> WhenConsRequirForTransferGivenNoRefuse { get; set; }
		//6.5
		public varClauseRef<bool> ISConsentIssueBankRequirTrans { get; set; }
		//6.6
		public varClauseRef<string> ExceptionalCircRelevInf { get; set; }
		//6.7
		public varNum<string> minTransAmount { get; set; }
		//6.8
		public varClauseRef<int> LendersGrantSecurOverOwnRight { get; set; }

		//7.1 
		public varClauseRef<string> PermiteedInterPeriod { get; set; }
		//7.2
		public varClauseRef<string> ConsentRequirNonStandardInterPeriod { get; set; }
		//7.3
		public varClauseRef<string> NoticePerForSubUtilRequestSterling { get; set; }
		//7.4 
		public varClauseRef<string> NoticePerForSubUtilRequestNonSterling { get; set; }
		//7.5
		public varNum<string> NotePerForRepayLoan { get; set; }
		//7.6
		public varNum<string> NotePerForCancelCommit { get; set; }


		//8.1
		public varNum<string> InitPercMargin { get; set; }
		//8.2
		public varClauseRef<bool> AreCommitmentFees { get; set; }
		//8.3
		public varClauseRef<bool> AreUtilFees { get; set; }
		//8.4
		public varClauseRef<bool> VaryMargin { get; set; }


		//9.1
		public varClauseRef<bool> CreditAgreeContainConcDefaultLenders { get; set; }
		//9.2
		public varClauseRef<bool> CanObligorsOffAgainstOwedDefaultLenders { get; set; }


		//10.1
		public string AgentTypes { get; set; }


		//11.1
		public string OtherInfo { get; set; }
		public DocDetail()
		{
			DefaultnoDefault = new List<EventPermit>(2);
			Common_Venders = new List<bool>(30);
			Commit = new List<CommitmentModel>(1);
			Contacts = new List<AOContact>(1);
		}
	}
}

File: Loan_Portfolio/Scripts/DocDetail.js
function AOContact()
	{

		this.FullName="";
		this.Role="";
		this.Contact="";
	}
	function CommitmentModel()
	{

		this.Facility="";
		this.Currency="";
		this.Amount=0;


	}

	function EventPermit()
	{
		this.TransferOtherLend =new varClauseRef("");
		this.DisclosureOtherLend =new varClauseRef("");
		this.TransferBankNFinancialInst =new varClauseRef("");
		this.DisclosureBankNFinancialInst =new varClauseRef("");

		//e
		this.TransferAnyEntitMakesLoans =new varClauseRef("d");

		//f
		this.DisclosureAnyEntitMakesLoans =new varClauseRef("d");
		this.AnyRequirPotentTransfereeRegulate =new varClauseRef("d");

	}


	function FacilityModel()
	{
		this.Type="";
		this.Maturity="";
		this.Facility="";
		this.Currency="";
		this.Amount=0;

	}
	function SecurityModel()
	{

		this.p1="";
		this.p2="";
		this.p3="";
		this.p4="";
		this.p5="";
		this.Comment="";

	}
	function varClauseRef(i)
	{
		this.ClauseRef="";
		this.OtherVar=i;
	}

	function varString(i)
	{
		this.STRING="";
		this.Bool=new varClauseRef(i);
	}
	function varNum(i)
	{
		this.Number=0.0;
		this.Bool=new varClauseRef(i);
	}
	function FinancialCovenantsModel()
	{
		this.debtLeverage =new varClauseRef(false);
		this.interestCover =new varClauseRef(false);
		this.CashflowCover =new varClauseRef(false);
		this.tangibleNetWorth =new varClauseRef(false);
	}


	function ObligationCauseAgentModel()
	{
		this.DocByBorrowToCreditors =new varClauseRef(false);
		this.DetailLitArbAdminProcess =new varClauseRef(false);
		this.NotificAnyDefault =new varClauseRef(false);
		this.FurtherInfoRequestFinConBusOp =new varClauseRef(false);
		this.NotificChngeOwnerOfBorrow =new varClauseRef(false);
		this.NotificChngeConstDocObligor =new varClauseRef(false);
		//ones with day input
		this.ConsAnnualFinancialStatement =new varNum(false);
		this.ConsSemiAnnualFinancialStatement =new varNum(false);
		this.ConsQuartAnnualFinancialStatement =new varNum(false);
		this.ConsMonthAnnualFinancialStatement =new varNum(false);
		this.UnConsAnnualFinancialStatement =new varNum(false);
		this.UnConsSemiAnnualFinancialStatement =new varNum(false);
		this.UnConsQuartAnnualFinancialStatement =new varNum(false);
		this.UnConsMonthAnnualFinancialStatement =new varNum(false);
	}


	function FormIdentificationModel()
	{

		this.Principle_Borrow="";
		this.AllenOveryCorM_Num="";
		this.Relevant_Facility="";
	}

	function SettingsModel()
	{
		this.Docname="";
		this.status="";
	}
	function DocDetail()
	{
		
		//settings
		this.settings=new SettingsModel();

		//security
		this.Security=new SecurityModel();


		//1.1
		 this.FormIdentification = new FormIdentificationModel();
		//1.2
		this.Common_Venders = new Array(30);
		for(var i=0; i<this.Common_Venders.length;i++){
			this.Common_Venders[i]=false;
		}
		this.AdditionalLenders="";
		this.final_report_dist_list="";
		//1.3
		this.PSponsor="";
		//1.4
		this.OrigDate="";
		//1.5
		this.Date="";
		//1.6
		this.Commit=[];
		this.Commit.push(new CommitmentModel())


		//1.7
		this.Contacts=[];
		this.Contacts.push(new AOContact());
		//1.8

		this.restricted=false;


		//2.1
		this.Facilities=new FacilityModel();
		//2.2 
		this.PurposeClause =new varClauseRef("");
		//2.3
		this.consentlvl="";
		//2.4
		this.extension="";
		//2.5
		this.CreditToInterestRate =new varClauseRef(false);
		//2.6
		this.DisposeMandatorPrepayment =new varClauseRef(false);
		//2.7
		this.ChangeMandatorPrepayment =new varClauseRef(false);
		//2.8
		this.CreditToTax_Gross_up =new varClauseRef(false);
		//2.9
		this.CreditToNegPledge =new varClauseRef(false);
		//2.10
		this.CreditToRestriction_IncurrringIndebtedness =new varClauseRef(false);
		//2.11
		this.FinancialCovenants=new FinancialCovenantsModel();
		//2.12
		this.MoreFinancialCovenants =new varClauseRef(false);
		//2.13
		this.ObligationCause=new ObligationCauseAgentModel();
		//2.14
		this.IsFailPayEvent=false;
		//2.15
		this.MaterialAdverseChangeEvent =new varClauseRef(false);
		//2.16
		this.ProvCrossDefault_CrossAccel =new varClauseRef("");
		//2.17
		this.CrossDefOrAccelThresh = new varString("");
		//2.18
		this.CAgrContClauseFavBank =new varClauseRef(false);
		//2.19
		this.GovLawPrincCAgr="";
		//2.20
		this.CAPArbitration =new varClauseRef(false);


		//3.1
		this.VoteOnPerameter =new varClauseRef(false);
		//3.2
		this.LenderSplitCommitLoansVote =new varClauseRef("");
		//3.3
		this.SplitVoteDetail =new varClauseRef("");
		//3.4
		this.HedgeBeforeEnf =new varClauseRef(false);
		//3.5
		this.HedgeAfterEnf =new varClauseRef(false);


		//4.1
		this.ObligorNameJurJob =new varClauseRef("");
		//4.2
		this.GuarantorsGuarantAllBorrow =new varClauseRef(false);
		//4.3
		this.DoesAgreeAllowAddBorrow =new varClauseRef(false);
		//4.4
		this.ConsLvlAdditionBorrowersNewJur =new varClauseRef("");
		//4.5
		this.DoesAgreeAllowAddGuarantors =new varClauseRef(false);
		//4.6
		this.ConsLvlAdditionBorrowersExstJur =new varClauseRef("");
		//4.7
		this.ConsLvlAdditionGuarantorsExstJur =new varClauseRef("");
		//4.8
		this.ConsLvlAdditionGuarantorsNewJur =new varClauseRef("");



		//5.1
		this.IsSecurity=false;
		//5.2
		this.SecurityDescrip="";
		//5.3
		this.LawsGovSecurity="";
		//5.4
		this.IsIntercreditorAgreement=false;
		//5.5
		this.SecurityBenefitHedge=false;
		//5.6
		this.IsOriginalDocProvidedToSecurityTrust=false;


		//6.1,6.2
		this.DefaultnoDefault=[];
		this.DefaultnoDefault.push(new EventPermit());
		this.DefaultnoDefault.push(new EventPermit());
		//6.3
		this.PotentTransfereeRequirEntNondisclosureAgree =new varClauseRef(false);
		//6.4
		this.WhenConsRequirForTransferGivenNoRefuse =new varClauseRef(false);
		//6.5
		this.ISConsentIssueBankRequirTrans =new varClauseRef(false);
		//6.6
		this.ExceptionalCircRelevInf =new varClauseRef("");
		//6.7
		this.minTransAmount =new varNum("");
		//6.8
		this.LendersGrantSecurOverOwnRight =new varClauseRef(0);

		//7.1 
		this.PermiteedInterPeriod =new varClauseRef("");
		//7.2
		this.ConsentRequirNonStandardInterPeriod =new varClauseRef("");
		//7.3
		this.NoticePerForSubUtilRequestSterling =new varClauseRef("");
		//7.4 
		this.NoticePerForSubUtilRequestNonSterling =new varClauseRef("");
		//7.5
		this.NotePerForRepayLoan =new varNum("");
		//7.6
		this.NotePerForCancelCommit =new varNum("");


		//8.1
		this.InitPercMargin =new varNum("");
		//8.2
		this.AreCommitmentFees =new varClauseRef(false);
		//8.3
		this.AreUtilFees =new varClauseRef(false);
		//8.4
		this.VaryMargin =new varClauseRef(false);


		//9.1
		this.CreditAgreeContainConcDefaultLenders =new varClauseRef(false);
		//9.2
		this.CanObligorsOffAgainstOwedDefaultLenders =new varClauseRef(false);


		//10.1
		this.AgentTypes="";

		//11.1
		this.OtherInfo="";
	}
